### PR TITLE
[Runtime] Handle dynamic casting to NSObject via error bridging.

### PR DESCRIPTION
**Pull request by DougGregor**
The dynamic casting machinery failed to account for `NSError` bridging when
casting to `NSObject`; check for it explicitly.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by mikeash** [comment]
Looks good. Can haz test?

**Comment by DougGregor** [comment]
Oh, I forgot to commit the test. Coming soon

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: test/stdlib/ErrorBridged.swift [hunk 1]
```diff
@@ -767,4 +767,29 @@ ErrorBridgingTests.test("@objc error domains for nested types") {
               String(reflecting: NonPrintAsObjCError.self))
 }
 
+@inline(never)
+@_optimize(none)
+func anyToAny<T, U>(_ a: T, _ : U.Type) -> U {
```
Comment (mikeash) [inline test/stdlib/ErrorBridged.swift]: This doesn't seem to be used. Is it vestigial or is there supposed to be more testing that uses it?

Comment (DougGregor) [reply test/stdlib/ErrorBridged.swift]: Its vestigial. Can you tell I did this late at night? 🤦‍♂

Comment (mikeash) [reply test/stdlib/ErrorBridged.swift]: If that's the worst thing that happened then I'd say you're doing pretty well.

Comment (DougGregor) [reply test/stdlib/ErrorBridged.swift]: https://github.com/apple/swift/pull/28357

**Pull request merged**